fitbenchmarking/controllers/base_controller.py

```diff
@@ -129,6 +130,27 @@ def eval_chisq(self, params, x=None, y=None, e=None):
         out = self.problem.eval_r_norm(params=params, x=x, y=y, e=e)
         return out
 
+    def check_attributes(self):
+        """
+        A helper function which checks all required attributes are set
+        in software controllers
+        """
+        values = {'flag': int}
+
+        for attr_name, attr_type in values.items():
+            attr = getattr(self, attr_name)
+            if not isinstance(attr, attr_type):
+                raise TypeError('Attribute "{}" is not the expected type.'
```

Comment:
Can you change the error to something like `Attribute "{}" in the controller is not the expected type.`, just to give the user (who's probably not familiar with the code base) a hint of where the error is.

Reply:
Done